crates/uv-configuration/src/dev.rs

```diff
@@ -1,445 +1,350 @@
-use std::borrow::Cow;
-
-use either::Either;
+use std::{borrow::Cow, sync::Arc};
 
 use uv_normalize::{GroupName, DEV_DEPENDENCIES};
 
-#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
-pub enum DevMode {
-    /// Include development dependencies.
-    #[default]
-    Include,
-    /// Exclude development dependencies.
-    Exclude,
-    /// Only include development dependencies, excluding all other dependencies.
-    Only,
-}
-
-impl DevMode {
-    /// Returns `true` if the specification allows for production dependencies.
-    pub fn prod(&self) -> bool {
-        matches!(self, Self::Exclude | Self::Include)
-    }
-
-    /// Returns `true` if the specification only includes development dependencies.
-    pub fn only(&self) -> bool {
-        matches!(self, Self::Only)
-    }
-
-    /// Returns the flag that was used to request development dependencies.
-    pub fn as_flag(&self) -> &'static str {
-        match self {
-            Self::Exclude => "--no-dev",
-            Self::Include => "--dev",
-            Self::Only => "--only-dev",
-        }
-    }
-
-    /// Returns `true` if the group is `dev`, and development dependencies should be included.
-    pub fn contains(&self, group: &GroupName) -> bool {
-        match self {
-            DevMode::Exclude => false,
-            DevMode::Include | DevMode::Only => group == &*DEV_DEPENDENCIES,
-        }
-    }
-}
-
-#[derive(Default, Debug, Clone)]
-pub struct DevGroupsSpecification {
-    /// Legacy option for `dependency-groups.dev` and `tool.uv.dev-dependencies`.
-    ///
-    /// Requested via the `--dev`, `--no-dev`, and `--only-dev` flags.
-    dev: Option<DevMode>,
+/// Manager of all dependency-group decisions and settings history.
+///
+/// This is an Arc mostly just to avoid size bloat on things that contain these.
+#[derive(Debug, Default, Clone)]
+pub struct DevGroupsSpecification(Arc<DevGroupsSpecificationInner>);
 
-    /// The groups to include.
+/// Manager of all dependency-group decisions and settings history.
+#[derive(Debug, Default, Clone)]
+pub struct DevGroupsSpecificationInner {
```

Comment:
I think this was called `DevGroups` because it included `dev` and `groups`? Also could be a leftover from the `dev` / `prod` split. Regardless, we can rename to `DependencyGroupsSpecification` or something.

Reply:
I'd prefer to do those kinds of renames in a followup PR, otherwise this one will be horribly merge-conflicty with non-trivial changes.

Reply:
Yep no problem